### PR TITLE
Place service rest implementation

### PR DESCRIPTION
**Pull request by cli1903**

### CODE REVIEW

File: frontend/ui/test/services/rest_api_place_visit_service_test.dart [hunk 1]
```diff
@@ -0,0 +1,133 @@
+import 'package:flutter_test/flutter_test.dart';
+import 'package:tripmeout/model/place_visit.dart';
+import 'package:tripmeout/services/rest_api_place_visit_service.dart';
+
+void main() {
+  group('Populated RestApiPlaceVisitService', () {
+    var placeVisitService =
+        RestApiPlaceVisitService(endpoint: 'http://localhost:8080');
```
Comment (bsidhom) [inline frontend/ui/test/services/rest_api_place_visit_service_test.dart:16]: Please only conditionally enable this test as in the [trip REST service test](https://github.com/googleinterns/step23-2020/blob/4223ce96c50ccae0d646f38e9e77c3153415bf3b/frontend/ui/test/services/rest_api_trip_service_test.dart#L8).

File: frontend/ui/lib/services/rest_api_place_visit_service.dart [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+  }
+
+  Future<PlaceVisit> createPlaceVisit(PlaceVisit placeVisit) async {
+    String tripid = placeVisit.tripid;
+    final http.Response response = await http.post(
+      '$endpoint/api/trips/$tripid/placeVisits',
+      headers: <String, String>{
+        'Content-Type': 'application/json; charset=UTF-8',
+      },
+      body: jsonEncode(<String, dynamic>{
```
Comment (bsidhom) [inline frontend/ui/lib/services/rest_api_place_visit_service.dart]: Do we need to manually construct the serializable Map here? We should move this logic so it lives alongside our deserialization code. Doing this inline (in two separate places) means that we have to keep both of these implementations in sync with the deserializer.

File: frontend/ui/lib/services/rest_api_place_visit_service.dart [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    }
+  }
+
+  Future<PlaceVisit> updatePlaceVisitUserMark(PlaceVisit placeVisit) async {
+    String tripid = placeVisit.tripid;
+    String id = placeVisit.id;
+    final http.Response response = await http.put(
+      '$endpoint/api/trips/$tripid/placeVisits/$id',
+      headers: <String, String>{
+        'Content-Type': 'application/json; charset=UTF-8',
```
Comment (bsidhom) [inline frontend/ui/lib/services/rest_api_place_visit_service.dart]: Manually entering these header values in-line is error-prone. This should at least be refactored into a shared constant.

File: frontend/ui/lib/model/place_visit.dart [hunk 4]
```diff
@@ -8,20 +8,15 @@ enum UserMark {
...
-  };
-
-  static UserMark userMarkEnumFromString(String userMark) {
-    UserMark mark = stringToUserMark[userMark];
-    if (mark == null) {
-      throw Exception("invalid user mark");
-    }
-    return mark;
-  }
+  static final Map<String, UserMark> stringToUserMark = Map.fromIterable(
```
Comment (KoopaKing) [inline frontend/ui/lib/model/place_visit.dart:11]: ngl this is kinda weird compared to just enumerating the string values. You have now further coupled the PlaceVisit model object to the wire format somewhere other than the `fromJson` method.

**Review by KoopaKing — APPROVED** [review]

**Review by anfee1 — APPROVED** [review]
LGTM

**Pull request merged**